mkosi/__init__.py

```diff
@@ -6791,24 +6786,62 @@ def cache_image_path(config: MkosiConfig, is_final_image: bool) -> Optional[Path
         return Path(f"{config.output}.{suffix}")
 
 
-def check_output(config: MkosiConfig) -> None:
+def check_tree_input(path: Optional[Path]) -> None:
+    # Each path may be a directory or a tarball.
+    # Open the file or directory to simulate an access check.
+    # If that fails, an exception will be thrown.
+    if not path:
+        return
+
+    os.open(path, os.R_OK)
+
+
+def check_script_input(path: Optional[Path]) -> None:
+    if not path:
+        return
+
+    os.open(path, os.R_OK)
+    if not path.is_file():
+        raise OSError(errno.ENOENT, 'Not a normal file')
+    if not os.access(path, os.X_OK):
+        raise OSError(errno.ENOENT, 'Not executable')
+    return None
+
+
+def check_inputs(config: MkosiConfig) -> None:
+    try:
+        for tree in (config.skeleton_trees,
+                     config.extra_trees):
```

Comment:
```suggestion
        for tree in (config.skeleton_trees, config.extra_trees):
```
since the line is very short already

Reply:
Hmm, this is on separate lines on purpose, to match the style of following loop. They both have lists of config options to go over, and I think it's nice to have the same structure.